omigami/plotting.py

```diff
@@ -0,0 +1,60 @@
+from PIL.PngImagePlugin import PngImageFile
+from pandas import DataFrame
+from rdkit import Chem
+from rdkit.Chem import Draw
+
+
+def _raise_errors(spectra_matches: DataFrame, representation: str = 'smiles'):
+    if representation not in ["smiles", "inchi"]:
+        raise ValueError(
+            f"Got unexpected representation string. Needs to be either 'smiles' or 'inchi' got {representation}"
+        )
+
+    if not isinstance(spectra_matches, type(DataFrame())):
+        raise ValueError(
+            f"Matches need to be a Pandas Dataframe got {type(spectra_matches)}"
+        )
+
+
+def _clean_matches(spectra_matches: DataFrame, representation: str) -> DataFrame:
+    spectra_matches = spectra_matches.drop_duplicates('compound_name')
+    spectra_matches = spectra_matches.drop_duplicates(representation)
+    spectra_matches = spectra_matches[spectra_matches[representation] != ""]
+    spectra_matches = spectra_matches[spectra_matches[representation].notna()]
+
+    return spectra_matches
+
+
+def plot_molecule_structure_grid(spectra_matches: DataFrame, representation: str = 'smiles',
```

Comment:
if they belonged to a class, you could move this function to the top and then it would increase readability a lot. Imagine, I started reading this file and it starts with a `_raise_errors` function.. I was like 🤔